syntax = "proto3";

package bluetooth.hal;

import "google/protobuf/empty.proto";

service HciHalFacade {
  rpc SendHciCommand(HciCommandPacket) returns (google.protobuf.Empty) {}
  rpc SendHciAcl(HciAclPacket) returns (google.protobuf.Empty) {}
  rpc SendHciSco(HciScoPacket) returns (google.protobuf.Empty) {}

  rpc FetchHciEvent(google.protobuf.Empty) returns (stream HciEventPacket) {}
  rpc FetchHciAcl(google.protobuf.Empty) returns (stream HciAclPacket) {}
  rpc FetchHciSco(google.protobuf.Empty) returns (stream HciScoPacket) {}
}

message HciEventPacket {
  bytes payload = 1;
}

message HciCommandPacket {
  bytes payload = 1;
}

message HciAclPacket {
  bytes payload = 1;
}

message HciScoPacket {
  bytes payload = 1;
}
